backend/src/main/java/ai/verta/modeldb/utils/ModelDBUtils.java

```diff
@@ -592,6 +595,60 @@ public static void checkPersonalWorkspace(
     }
   }
 
+  public static void checkPersonalWorkspace(
+      UserInfo userInfo, Workspace workspace, String resourceNameString) {
+    if (userInfo != null
+        && workspace.getInternalIdCase() == Workspace.InternalIdCase.USER_ID
+        && !workspace.getUserId().equals(userInfo.getVertaInfo().getUserId())) {
+      Status status =
+          Status.newBuilder()
+              .setCode(Code.PERMISSION_DENIED_VALUE)
+              .setMessage(
+                  "Creation of "
+                      + resourceNameString
+                      + " in other user's workspace is not permitted")
+              .addDetails(Any.pack(UpdateProjectName.Response.getDefaultInstance()))
+              .build();
+      throw StatusProto.toStatusRuntimeException(status);
+    }
+  }
```

Comment:
Can't we just let UAC handle this?

Reply:
No, this logic is just check the requested workspace is matched with current login users workspace and we have enough info to check that instead of calling UAC.

Reply:
It's not a matter of whether we can or not, but whether we should or not. Every time we make a permission decision in a component vs UAC, it needs to be duplicated in exactly the same way in all components, which is much more error prone. It would be much more robust to have this permission verified in a central place that every system can rely on, wouldn't you agree?

Reply:
Yes, removing from MDB and i hope the UAC handle this thing on the `serResource` endpoint

Reply:
If it doesn't, I'd say it's a UAC bug that we have to fix anyway. cc @evgeniilezhnin to double check